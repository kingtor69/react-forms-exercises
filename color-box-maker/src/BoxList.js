import React, { useState } from 'react';
import NewBoxForm from './NewBoxForm.js';
import Box from './Box.js';

const BoxList = () => {
  const initialBoxState = [
    // {
    //   color: "blue",
    //   width: 150,
    //   height: 90
    // }
  ];
  const [ boxes, setBoxes ] = useState(initialBoxState);
  const makeNewBox = (newBoxData) => {
    setBoxes(boxes => [...boxes, newBoxData])
  };

  const removeBox = (boxId) => {
    setBoxes(boxes.filter ((b, i) => (i !== boxId)));
  };

  return (
    <div className="NewBoxList">
      <NewBoxForm 
        setBoxes = {setBoxes}
        makeNewBox={makeNewBox}
      />
      {boxes.map((box, i) => (
        <Box 
          color={box.color}
          width={box.width}
          height={box.height}
          id={i}
          key={i}
          removeBox={removeBox}
        />)
      )}
    </div>
  );
};

export default BoxList;
