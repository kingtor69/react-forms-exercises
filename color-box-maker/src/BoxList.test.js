import { render } from '@testing-library/react';
import BoxList from './BoxList';

it('renders witout ceashing', () => {
  render(<BoxList />);
});

it('renders snapshot', () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});
