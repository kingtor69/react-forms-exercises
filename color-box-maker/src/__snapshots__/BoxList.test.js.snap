// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewBoxList"
  >
    <div
      class="NewBoxForm"
    >
      <form
        class="form"
      >
        <div
          class="form-group"
        >
          <label
            for="color"
          >
            Color:
          </label>
          <input
            id="color"
            name="color"
            placeholder="color"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="width"
          >
            Width:
          </label>
          <input
            id="width"
            name="width"
            placeholder="width"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="height"
          >
            Height:
          </label>
          <input
            id="height"
            name="height"
            placeholder="height"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <button
            type="submit"
          >
            make new box
          </button>
        </div>
      </form>
    </div>
    <div
      class="Box"
      data-testid="box0"
      style="background-color: blue; width: 150px; height: 90px;"
    />
  </div>
</DocumentFragment>
`;
